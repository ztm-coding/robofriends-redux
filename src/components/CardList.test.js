import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import CardList from "./CardList";

// Mock data
const mockRobots = [
    { id: 1, name: "John Doe", email: "john@example.com" },
    { id: 2, name: "Jane Doe", email: "jane@example.com" },
];

describe("CardList Component", () => {
    test("renders the correct number of Card components", () => {
        const { asFragment } = render(<CardList robots={mockRobots} />);
        // Get all Card components by the test ID
        const cards = screen.getAllByTestId("card-component");

        // Assert that the number of rendered cards matches the length of mockRobots
        expect(cards).toHaveLength(mockRobots.length);
        expect(asFragment()).toMatchSnapshot();
    });

    test("displays the correct names and emails", () => {
        const { asFragment } = render(<CardList robots={mockRobots} />);

        // Check if each robot's name and email is displayed
        mockRobots.forEach((robot) => {
            expect(screen.getByText(robot.name)).toBeInTheDocument();
            expect(screen.getByText(robot.email)).toBeInTheDocument();
        });
        expect(asFragment()).toMatchSnapshot();
    });

    test("renders nothing when robots array is empty", () => {
        const { asFragment } = render(<CardList robots={[]} />);

        // There should be no Card components when the array is empty
        expect(screen.queryByTestId("card-component")).not.toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});
