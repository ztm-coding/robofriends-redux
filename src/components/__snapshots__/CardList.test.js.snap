// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component displays the correct names and emails 1`] = `
<DocumentFragment>
  <div>
    <div
      class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
      data-testid="card-component"
    >
      <img
        alt="robots"
        height="200"
        src="https://robohash.org/1?size=200x200"
        width="200"
      />
      <div>
        <h2>
          John Doe
        </h2>
        <p>
          john@example.com
        </p>
      </div>
    </div>
    <div
      class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
      data-testid="card-component"
    >
      <img
        alt="robots"
        height="200"
        src="https://robohash.org/2?size=200x200"
        width="200"
      />
      <div>
        <h2>
          Jane Doe
        </h2>
        <p>
          jane@example.com
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardList Component renders nothing when robots array is empty 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`CardList Component renders the correct number of Card components 1`] = `
<DocumentFragment>
  <div>
    <div
      class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
      data-testid="card-component"
    >
      <img
        alt="robots"
        height="200"
        src="https://robohash.org/1?size=200x200"
        width="200"
      />
      <div>
        <h2>
          John Doe
        </h2>
        <p>
          john@example.com
        </p>
      </div>
    </div>
    <div
      class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
      data-testid="card-component"
    >
      <img
        alt="robots"
        height="200"
        src="https://robohash.org/2?size=200x200"
        width="200"
      />
      <div>
        <h2>
          Jane Doe
        </h2>
        <p>
          jane@example.com
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
